pkg/backupschedule/createManager.go

```diff
@@ -0,0 +1,125 @@
+package backupschedule
+
+import (
+	"context"
+	"fmt"
+
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	v1 "github.com/cloudogu/k8s-backup-operator/pkg/api/v1"
+	"github.com/cloudogu/k8s-backup-operator/pkg/retry"
+)
+
+type defaultCreateManager struct {
+	clientSet ecosystemInterface
+	recorder  eventRecorder
+	namespace string
+}
+
+func newCreateManager(clientSet ecosystemInterface, recorder eventRecorder, namespace string) *defaultCreateManager {
+	return &defaultCreateManager{clientSet: clientSet, recorder: recorder, namespace: namespace}
+}
+
+func (cm *defaultCreateManager) create(ctx context.Context, backupSchedule *v1.BackupSchedule) error {
+	cm.recorder.Event(backupSchedule, corev1.EventTypeNormal, v1.CreateEventReason, "Creating backup schedule")
+
+	backupScheduleClient := cm.clientSet.EcosystemV1Alpha1().BackupSchedules(cm.namespace)
+
+	backupScheduleName := backupSchedule.Name
+	backupSchedule, err := backupScheduleClient.UpdateStatusCreating(ctx, backupSchedule)
+	if err != nil {
+		return fmt.Errorf("failed to set status [%s] in backup schedule resource [%s]: %w", v1.BackupScheduleStatusCreating, backupScheduleName, err)
+	}
+
+	backupSchedule, err = backupScheduleClient.AddFinalizer(ctx, backupSchedule, v1.BackupScheduleFinalizer)
+	if err != nil {
+		return fmt.Errorf("failed to add finalizer [%s] in backup schedule resource [%s]: %w", v1.BackupScheduleFinalizer, backupScheduleName, err)
+	}
+
+	backupSchedule, err = backupScheduleClient.AddLabels(ctx, backupSchedule)
+	if err != nil {
+		return fmt.Errorf("failed to add labels to backup schedule resource [%s]: %w", backupScheduleName, err)
+	}
+
+	err = cm.createCronJob(ctx, backupSchedule)
+	if err != nil {
+		return fmt.Errorf("failed to create cron job for backup schedule [%s]: %w", backupScheduleName, err)
+	}
+
+	_, err = backupScheduleClient.UpdateStatusCreated(ctx, backupSchedule)
+	if err != nil {
+		return fmt.Errorf("failed to set status [%s] in backup schedule resource [%s]: %w", v1.BackupScheduleStatusCreated, backupScheduleName, err)
+	}
+
+	return nil
+}
+
+func (cm *defaultCreateManager) createCronJob(ctx context.Context, schedule *v1.BackupSchedule) error {
```

Comment:
🚫 **[checkstyle]** <funlen> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Function 'createCronJob' is too long (65 > 60)